### PR TITLE
(PUP-10429) Allow puppetserver to register http implementation

### PR DESCRIPTION
**Pull request by joshcooper**
Allow puppetserver to register its http implementation

### CODE REVIEW

**Comment by joshcooper** [comment]
Hey @justinstoller @tvpartytonight, does this look like what you're thinking? I was surprised to find that puppetserver sets the http_client_class before initializing settings.

**Comment by justinstoller** [comment]
I think that passing this into `initialize_settings` or into some kind of system boostrapping method is the desired end goal still, right? Assuming so, I don't think the current weirdness in Server should be a determining factor.

There's a fair bit in the Server bootstrapping of Puppet that could be done at different times but I don't want to change them w/o reason. Looking on the commit that separated out the Server's Config and its PuppetConfig, it looks like the HttpClient used to be set after settings were initialized. It was moved into the server's config which is called first w/o comment in: https://github.com/puppetlabs/puppetserver/commit/9ebd9603f483856ba4b6d274aee5a6719c222fc7

And aside from that question, I think this all looks great. Thanks, Josh!

(ps. Tony's out on paternity leave)

File: spec/unit/network/http_pool_spec.rb [hunk 1]
```diff
@@ -1,12 +1,41 @@
...
+  attr_reader :host, :port
+end
+
 describe Puppet::Network::HttpPool do
   before :each do
     Puppet::SSL::Key.indirection.terminus_class = :memory
     Puppet::SSL::CertificateRequest.indirection.terminus_class = :memory
   end
 
+  describe "" do
```
Comment (melissa) [inline spec/unit/network/http_pool_spec.rb:18]: empty describe block

**Comment by melissa** [comment]
Just one nit, otherwise it looks good

File: spec/unit/network/http_pool_spec.rb (hunk 1 above)
Comment (joshcooper) [reply spec/unit/network/http_pool_spec.rb:18]: haha nice!

**Comment by joshcooper** [comment]
@justinstoller yeah I think `initialize_settings` is fine. I made some updates to the tests, and added you as a reviewer.

**Pull request merged**